### PR TITLE
feat: delete item thumbnail

### LINKED ISSUE #1155: Allow item thumbnail deletion

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #1155 

Screenshot from 2024-04-18 15-39-54

### CODE REVIEW

File: src/components/item/settings/ThumbnailWithDeleteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+const ThumbnailWithDeleteButton = ({
+  itemId,
+  isLoading,
+  url,
+  alt,
+  onDelete,
+  hasThumbnail,
+}: Props): JSX.Element => {
+  const [isHovered, setIsHovered] = useState(false); // State to manage hover
```
Comment (pyphilia) [inline src/components/item/settings/ThumbnailWithDeleteButton.tsx:36]:
```suggestion
  const [isHovered, setIsHovered] = useState(false);
```

File: package.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
     "@emotion/styled": "11.11.5",
     "@graasp/chatbox": "3.1.0",
     "@graasp/map": "1.10.0",
-    "@graasp/query-client": "3.4.1",
+    "@graasp/query-client": "https://github.com/graasp/graasp-query-client.git#763-delete-thumbnail-query",
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply package.json:24]: I'll update after the new release :)

File: cypress/e2e/item/settings/thumbnail.cy.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        .should('be.visible');
+      cy.get(`#${ITEM_THUMBNAIL_DELETE_BTN_ID}`).click();
+      cy.wait(`@deleteItemThumbnail`).then(({ request: { url } }) => {
+        expect(url).to.contain(items[1].id);
+      });
+    });
+
+    it('Delete thumbnail button should not exist for item with no thumbnail', () => {
+      cy.visit(buildItemSettingsPath(SAMPLE_ITEMS.items[4].id));
+      cy.get(`#${ITEM_THUMBNAIL_CONTAINER_ID}`).trigger('mouseover');
```
Comment (pyphilia) [inline cypress/e2e/item/settings/thumbnail.cy.ts:54]: You told me this doesn't work right? If you use `.invoke('show')` on a button that does not exist, does that throw? maybe you could catch that so you know the button does not exist.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Just a change for a test, and update the yarn lock, and that should be good!

**Pull request merged**